alembic/versions/ab83c84a325b_include_cascading_deletion.py

```diff
@@ -0,0 +1,86 @@
+"""include cascading deletion
+
+Revision ID: ab83c84a325b
+Revises: eaca44fbeabe
+Create Date: 2019-02-12 11:54:49.151206
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = 'ab83c84a325b'
+down_revision = 'eaca44fbeabe'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_constraint('blocks_office_id_fkey', 'blocks', type_='foreignkey')
+    op.create_foreign_key(None, 'blocks', 'offices', ['office_id'], ['id'], ondelete='CASCADE')
```

Comment:
Why did we add `ondelete='CASCADE'` here on the alembic file other than using the one on the model. I think the one on the model should work.

Reply:
The reason for this is that the cascade="all,delete" does not delete the children, it instead removes the relation by setting the foreign key as null. When I tried running the script without the changes the database threw an error that the foreign key constraint was infringed on. This was the solution I found after doing some research.

Reply:
@caveinn , Have you tried `all, delete-orphan` in the models as well?

Reply:
@RonKbS I tried that too. After reading the documentation I learned that the cascade attribute on its own does not delete the children it disassociates it. You have to couple them with the ondelete="cascade" to delete the children.